--- a/htdocs/luci-static/resources/view/network/wireless.js
+++ b/htdocs/luci-static/resources/view/network/wireless.js
@@ -950,6 +950,9 @@ return view.extend({
 					o.value('2', _('High'));
 					o.value('3', _('Very High'));
 
+					o = ss.taboption('advanced', form.Flag, 'mu_beamformer', _('MU-MIMO'));
+					o.default = o.enabled;
+
 					o = ss.taboption('advanced', form.Value, 'distance', _('Distance Optimization'), _('Distance to farthest network member in meters. Set only for distances above one kilometer; otherwise it is harmful.'));
 					o.datatype = 'or(range(0,114750),"auto")';
 					o.placeholder = 'auto';
@@ -969,6 +972,9 @@ return view.extend({
 					o.datatype = 'range(15,65535)';
 					o.placeholder = 100;
 					o.rmempty = true;
+
+					o = ss.taboption('advanced', form.Flag, 'vendor_vht', _('Enable 256-QAM'), _('Enable 256QAM modulation. May not supported by some hardware!'));
+					o.rmempty = true;
 				}
 
 
--- a/htdocs/luci-static/resources/view/network/interfaces.js
+++ b/htdocs/luci-static/resources/view/network/interfaces.js
@@ -889,6 +889,20 @@ return view.extend({
 					so.depends('ra', 'server');
 					so.depends({ ra: 'hybrid', master: '0' });
 
+					so = ss.taboption('ipv6-ra', form.Value, 'max_preferred_lifetime', _('Max preferred lifetime'));
+					so.optional = true;
+					so.datatype = 'range(0, 2700)';
+					so.placeholder = '2700';
+					so.depends('ra', 'server');
+					so.depends({ ra: 'hybrid', master: '0' });
+
+					so = ss.taboption('ipv6-ra', form.Value, 'max_valid_lifetime', _('Max valid lifetime'));
+					so.optional = true;
+					so.datatype = 'range(0, 5400)';
+					so.placeholder = '5400';
+					so.depends('ra', 'server');
+					so.depends({ ra: 'hybrid', master: '0' });
+
 					so = ss.taboption('ipv6-ra', form.Value, 'ra_maxinterval', _('Max <abbr title="Router Advertisement">RA</abbr> interval'), _('Maximum time allowed  between sending unsolicited <abbr title="Router Advertisement, ICMPv6 Type 134">RA</abbr>. Default is 600 seconds.'));
 					so.optional = true;
 					so.datatype = 'uinteger';
@@ -1006,18 +1020,6 @@ return view.extend({
 					so = ss.taboption('ipv6', form.Flag, 'ndproxy_slave', _('NDP-Proxy slave'), _('Set interface as NDP-Proxy external slave. Default is off.'));
 					so.depends({ ndp: 'relay', master: '0' });
 					so.depends({ ndp: 'hybrid', master: '0' });
-
-					so = ss.taboption('ipv6', form.Value, 'preferred_lifetime', _('IPv6 Prefix Lifetime'), _('Preferred lifetime for a prefix.'));
-					so.optional = true;
-					so.placeholder = '12h';
-					so.value('5m', _('5m (5 minutes)'));
-					so.value('3h', _('3h (3 hours)'));
-					so.value('12h', _('12h (12 hours - default)'));
-					so.value('7d', _('7d (7 days)'));
-
-					//This is a ra_* setting, but its placement is more logical/findable under IPv6 settings.
-					so = ss.taboption('ipv6', form.Flag, 'ra_useleasetime', _('Follow IPv4 Lifetime'), _('DHCPv4 <code>leasetime</code> is used as limit and preferred lifetime of the IPv6 prefix.'));
-					so.optional = true;
 				}
 
 				ifc.renderFormOptions(s);
