--- a/image/Makefile
+++ b/image/Makefile
@@ -497,7 +497,8 @@ define Device/phicomm_k3
   DEVICE_ALT0_MODEL := QUANTUM DAX
   DEVICE_ALT1_VENDOR := Wavlink
   DEVICE_ALT1_MODEL := WL-WN538A8
-  DEVICE_PACKAGES := $(BRCMFMAC_4366C0) $(USB3_PACKAGES)
+  DEVICE_PACKAGES := $(BRCMFMAC_4366C0) $(USB3_PACKAGES) \
+	-brcmfmac-firmware-4366c0-pcie brcmfmac-firmware-4366c0-pcie-k3
   IMAGES := trx
 endef
 TARGET_DEVICES += phicomm_k3
