diff --git a/config_generate b/config_generate
index f062c10..20f6bdf 100755
--- a/config_generate
+++ b/config_generate
@@ -311,10 +311,10 @@ generate_static_system() {
 		set system.ntp='timeserver'
 		set system.ntp.enabled='1'
 		set system.ntp.enable_server='0'
+		add_list system.ntp.server='time1.google.com'
+		add_list system.ntp.server='time1.apple.com'
 		add_list system.ntp.server='0.openwrt.pool.ntp.org'
 		add_list system.ntp.server='1.openwrt.pool.ntp.org'
-		add_list system.ntp.server='2.openwrt.pool.ntp.org'
-		add_list system.ntp.server='3.openwrt.pool.ntp.org'
 	EOF
 
 	if json_is_a system object; then
