diff --git a/files/aria2.init b/files/aria2.init
index deef307..0147d08 100755
--- a/files/aria2.init
+++ b/files/aria2.init
@@ -168,6 +168,11 @@ aria2_start() {
 	config_file_tmp="$config_dir/$NAME.conf.tmp"
 	session_file="$config_dir/$NAME.session.$section"
 
+	_make_dir "$dir" || {
+		_err "Can't create download dir: $dir"
+		return 1
+	}
+
 	_make_dir "$config_dir" || {
 		_err "Can't create config dir: $config_dir"
 		return 1
