diff --git a/files/lib/netifd/wireless/mac80211.sh b/files/lib/netifd/wireless/mac80211.sh
index d74a2e0..89f4090 100644
--- a/files/lib/netifd/wireless/mac80211.sh
+++ b/files/lib/netifd/wireless/mac80211.sh
@@ -419,7 +419,16 @@ mac80211_hostapd_setup_base() {
 		he_mac_cap=${he_mac_cap:2}
 
 		append base_cfg "ieee80211ax=1" "$N"
-		[ -n "$he_bss_color" ] && append base_cfg "he_bss_color=$he_bss_color" "$N"
+
+		if [ -n "$he_bss_color" ]; then
+			append base_cfg "he_bss_color=$he_bss_color" "$N"
+		else
+			he_bss_color=$(head -1 /dev/urandom | tr -dc '0-9' | head -c2)
+			he_bss_color=$(($he_bss_color % 63))
+			he_bss_color=$(($he_bss_color + 1))
+			append base_cfg "he_bss_color=$he_bss_color" "$N"
+		fi
+
 		[ "$hwmode" = "a" ] && {
 			append base_cfg "he_oper_chwidth=$vht_oper_chwidth" "$N"
 			append base_cfg "he_oper_centr_freq_seg0_idx=$vht_center_seg0" "$N"
