diff --git a/files/lib/wifi/mac80211.sh b/files/lib/wifi/mac80211.sh
index c0fbfbe..b9524fa 100644
--- a/files/lib/wifi/mac80211.sh
+++ b/files/lib/wifi/mac80211.sh
@@ -174,8 +174,9 @@ detect_mac80211() {
 			set wireless.radio${devidx}.channel=${channel}
 			set wireless.radio${devidx}.band=${mode_band}
 			set wireless.radio${devidx}.htmode=$htmode
-			set wireless.radio${devidx}.disabled=1
+			set wireless.radio${devidx}.disabled=0
 
+			set wireless.radio${devidx}.country=US
 			set wireless.default_radio${devidx}=wifi-iface
 			set wireless.default_radio${devidx}.device=radio${devidx}
 			set wireless.default_radio${devidx}.network=lan
