diff --git a/bridge/nss_bridge_mgr.c b/bridge/nss_bridge_mgr.c
index f202ad1..48d4a39 100644
--- a/bridge/nss_bridge_mgr.c
+++ b/bridge/nss_bridge_mgr.c
@@ -1081,7 +1081,7 @@ int nss_bridge_mgr_register_br(struct ne
 	}
 #endif

-	err = nss_bridge_tx_set_mac_addr_msg(ifnum, dev->dev_addr);
+	err = nss_bridge_tx_set_mac_addr_msg(ifnum, (uint8_t *) dev->dev_addr);
 	if (err != NSS_TX_SUCCESS) {
 		nss_bridge_mgr_warn("%px: failed to set mac_addr msg, error = %d\n", b_pvt, err);
 		goto fail_4;
@@ -1242,7 +1242,7 @@ static int nss_bridge_mgr_changeaddr_eve

 	nss_bridge_mgr_trace("%px: MAC changed to %pM, update NSS\n", b_pvt, dev->dev_addr);

-	if (nss_bridge_tx_set_mac_addr_msg(b_pvt->ifnum, dev->dev_addr) != NSS_TX_SUCCESS) {
+	if (nss_bridge_tx_set_mac_addr_msg(b_pvt->ifnum, (uint8_t *) dev->dev_addr) != NSS_TX_SUCCESS) {
 		nss_bridge_mgr_warn("%px: Failed to send change MAC address message to NSS\n", b_pvt);
 		return NOTIFY_DONE;
 	}
diff --git a/clmapmgr/nss_clmapmgr.c b/clmapmgr/nss_clmapmgr.c
index 9574706..c510032 100644
--- a/clmapmgr/nss_clmapmgr.c
+++ b/clmapmgr/nss_clmapmgr.c
@@ -103,7 +103,11 @@ static struct rtnl_link_stats64 *nss_clmapmgr_get_dev_stats64(struct net_device
 	 * Netdev seems to be incrementing rx_dropped because we don't give IP header.
 	 * So reset it as it's of no use for us.
 	 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	atomic_long_set(&dev->rx_dropped, 0);
+#else
+	atomic_long_set(&(dev)->stats.__rx_dropped, 0);
+#endif
 	priv = netdev_priv(dev);
 	memset(stats, 0, sizeof(struct rtnl_link_stats64));
 	memcpy(stats, &priv->stats, sizeof(struct rtnl_link_stats64));
diff --git a/dtls/v2.0/nss_dtlsmgr_ctx_dev.c b/dtls/v2.0/nss_dtlsmgr_ctx_dev.c
index 2bbe29b..40eccf0 100644
--- a/dtls/v2.0/nss_dtlsmgr_ctx_dev.c
+++ b/dtls/v2.0/nss_dtlsmgr_ctx_dev.c
@@ -532,7 +532,11 @@ void nss_dtlsmgr_ctx_dev_setup(struct net_device *dev)
 #else
 	dev->priv_destructor = nss_dtlsmgr_ctx_dev_free;
 #endif
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	memcpy(dev->dev_addr, "\xaa\xbb\xcc\xdd\xee\xff", dev->addr_len);
+#else
+	memcpy((void *) dev->dev_addr, "\xaa\xbb\xcc\xdd\xee\xff", dev->addr_len);
+#endif
 	memset(dev->broadcast, 0xff, dev->addr_len);
 	memcpy(dev->perm_addr, dev->dev_addr, dev->addr_len);
 }
diff --git a/gre/nss_connmgr_gre.c b/gre/nss_connmgr_gre.c
index 4797264..82a32da 100644
--- a/gre/nss_connmgr_gre.c
+++ b/gre/nss_connmgr_gre.c
@@ -279,10 +279,17 @@ static struct rtnl_link_stats64 *nss_connmgr_gre_get_dev_stats64(struct net_devi
 #else
 			start = u64_stats_fetch_begin_irq(&tstats->syncp);
 #endif
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 			rx_packets = tstats->rx_packets;
 			tx_packets = tstats->tx_packets;
 			rx_bytes = tstats->rx_bytes;
 			tx_bytes = tstats->tx_bytes;
+#else
+			rx_packets = u64_stats_read(&tstats->rx_packets);
+			tx_packets = u64_stats_read(&tstats->tx_packets);
+			rx_bytes = u64_stats_read(&tstats->rx_bytes);
+			tx_bytes = u64_stats_read(&tstats->tx_bytes);
+#endif
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(3, 15, 0))
 		} while (u64_stats_fetch_retry_bh(&tstats->syncp, start));
 #else
@@ -697,11 +704,21 @@ static void nss_connmgr_gre_event_receive(void *if_ctx, struct nss_gre_msg *tnlm
 		tstats = this_cpu_ptr(dev->tstats);
 		u64_stats_update_begin(&tstats->syncp);
 		if (interface_type == NSS_DYNAMIC_INTERFACE_TYPE_GRE_INNER) {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 			tstats->tx_packets += stats->tx_packets;
 			tstats->tx_bytes += stats->tx_bytes;
+#else
+			u64_stats_add(&tstats->tx_packets, stats->tx_packets);
+			u64_stats_add(&tstats->tx_bytes, stats->tx_bytes);
+#endif
 		} else if (interface_type == NSS_DYNAMIC_INTERFACE_TYPE_GRE_OUTER) {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 			tstats->rx_packets += stats->rx_packets;
 			tstats->rx_bytes += stats->rx_bytes;
+#else
+			u64_stats_add(&tstats->rx_packets, stats->rx_packets);
+			u64_stats_add(&tstats->rx_bytes, stats->rx_bytes);
+#endif
 		}
 		u64_stats_update_end(&tstats->syncp);
 		dev->stats.rx_dropped += nss_cmn_rx_dropped_sum(stats);
diff --git a/gre/test/nss_connmgr_gre_test.c b/gre/test/nss_connmgr_gre_test.c
index d77eeee..dbf9576 100644
--- a/gre/test/nss_connmgr_gre_test.c
+++ b/gre/test/nss_connmgr_gre_test.c
@@ -223,7 +223,11 @@ static int nss_connmgr_gre_test_show_proc(struct seq_file *filp, void *data)
  */
 static int nss_connmgr_gre_test_open_proc(struct inode *inode, struct file *filp)
 {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	return single_open(filp, nss_connmgr_gre_test_show_proc, PDE_DATA(inode));
+#else
+	return single_open(filp, nss_connmgr_gre_test_show_proc, pde_data(inode));
+#endif
 }
 
 /*
diff --git a/netlink/nss_nlgre_redir_cmn.c b/netlink/nss_nlgre_redir_cmn.c
index 2517622..fff9f04 100644
--- a/netlink/nss_nlgre_redir_cmn.c
+++ b/netlink/nss_nlgre_redir_cmn.c
@@ -384,7 +384,11 @@ static int nss_nlgre_redir_cmn_set_mac_address(struct net_device *dev, void *p)
 		return -EINVAL;
 	}
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	memcpy(dev->dev_addr, addr->sa_data, ETH_ALEN);
+#else
+	memcpy((void *) dev->dev_addr, addr->sa_data, ETH_ALEN);
+#endif
 	return 0;
 }
 
diff --git a/nss_connmgr_tun6rd.c b/nss_connmgr_tun6rd.c
index ffa6ad4..4c1e841 100644
--- a/nss_connmgr_tun6rd.c
+++ b/nss_connmgr_tun6rd.c
@@ -101,11 +101,18 @@ static void nss_tun6rd_update_dev_stats(struct net_device *dev,
 
 	u64_stats_init(&stats.syncp);
 	u64_stats_update_begin(&stats.syncp);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	stats.rx_packets = sync_stats->node_stats.rx_packets;
 	stats.rx_bytes = sync_stats->node_stats.rx_bytes;
 	stats.tx_packets = sync_stats->node_stats.tx_packets;
 	stats.tx_bytes = sync_stats->node_stats.tx_bytes;
+#else
+	u64_stats_set(&stats.rx_packets, sync_stats->node_stats.rx_packets);
+	u64_stats_set(&stats.rx_bytes, sync_stats->node_stats.rx_bytes);
+	u64_stats_set(&stats.tx_packets, sync_stats->node_stats.tx_packets);
+	u64_stats_set(&stats.tx_bytes, sync_stats->node_stats.tx_bytes);
 	u64_stats_update_end(&stats.syncp);
+#endif
 #else
 	struct nss_tun6rd_stats stats;
 
diff --git a/nss_qdisc/igs/nss_ifb.c b/nss_qdisc/igs/nss_ifb.c
index 7e20f23..eb82d27 100644
--- a/nss_qdisc/igs/nss_ifb.c
+++ b/nss_qdisc/igs/nss_ifb.c
@@ -544,8 +544,15 @@ static void nss_ifb_update_dev_stats(struct net_device *dev, struct nss_igs_stat
 	 * post shaping. Therefore IFB interface's stats should be updated
 	 * with NSS firmware's IFB TX stats only.
 	 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	stats.rx_packets = stats.tx_packets = node_stats->tx_packets;
 	stats.rx_bytes = stats.tx_bytes = node_stats->tx_bytes;
+#else
+	u64_stats_set(&stats.rx_packets, node_stats->tx_packets);
+	u64_stats_set(&stats.tx_packets, node_stats->tx_packets);
+	u64_stats_set(&stats.rx_bytes, node_stats->tx_bytes);
+	u64_stats_set(&stats.tx_bytes, node_stats->tx_bytes);
+#endif
 	dev->stats.rx_dropped = dev->stats.tx_dropped += sync_stats->igs_stats.tx_dropped;
 	u64_stats_update_end(&stats.syncp);
 
diff --git a/nss_qdisc/igs/nss_mirred.c b/nss_qdisc/igs/nss_mirred.c
index d447494..1019ba0 100644
--- a/nss_qdisc/igs/nss_mirred.c
+++ b/nss_qdisc/igs/nss_mirred.c
@@ -317,7 +317,11 @@ static int nss_mirred_act(struct sk_buff *skb, const struct tc_action *tc_act,
 	 * Update the last use of action.
 	 */
 	tcf_lastuse_update(&act->tcf_tm);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	bstats_cpu_update(this_cpu_ptr(act->common.cpu_bstats), skb);
+#else
+	bstats_update(this_cpu_ptr(act->common.cpu_bstats), skb);
+#endif
 
 	rcu_read_lock();
 	retval = READ_ONCE(act->tcf_action);
diff --git a/nss_qdisc/nss_qdisc.c b/nss_qdisc/nss_qdisc.c
index 1272277..ab48897 100644
--- a/nss_qdisc/nss_qdisc.c
+++ b/nss_qdisc/nss_qdisc.c
@@ -2189,9 +2189,9 @@ int __nss_qdisc_init(struct Qdisc *sch,
 	 * This is to prevent mixing NSS and PPE qdisc with linux qdisc.
 	 */
 	if ((parent != TC_H_ROOT) && (root->ops->owner != THIS_MODULE)) {
-		nss_qdisc_warning("parent (%d) and TC_H_ROOT (%d))", parent, TC_H_ROOT);
-		nss_qdisc_warning("root->ops->owner (%px) and THIS_MODULE (%px))", root->ops->owner , THIS_MODULE);
-		nss_qdisc_warning("NSS qdisc %px (type %d) used along with non-nss qdiscs,"
+		nss_qdisc_info("parent (%d) and TC_H_ROOT (%d))", parent, TC_H_ROOT);
+		nss_qdisc_info("root->ops->owner (%px) and THIS_MODULE (%px))", root->ops->owner , THIS_MODULE);
+		nss_qdisc_info("NSS qdisc %px (type %d) used along with non-nss qdiscs,"
 			" or the interface is currently down", nq->qdisc, nq->type);
 	}
 
@@ -2608,12 +2608,18 @@ int nss_qdisc_init(struct Qdisc *sch, struct nss_qdisc *nq, nss_shaper_node_type
  *  Wrapper around gnet_stats_copy_basic()
  */
 int nss_qdisc_gnet_stats_copy_basic(struct Qdisc *sch, struct gnet_dump *d,
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 				struct gnet_stats_basic_packed *b)
+#else
+				struct gnet_stats_basic_sync *b)
+#endif
 {
 #if (LINUX_VERSION_CODE <= KERNEL_VERSION(3, 18, 0))
 	return gnet_stats_copy_basic(d, b);
 #elif (LINUX_VERSION_CODE < KERNEL_VERSION(4, 8, 0))
 	return gnet_stats_copy_basic(d, NULL, b);
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 16, 0))
+	return gnet_stats_copy_basic(d, NULL, b, true);
 #else
 	return gnet_stats_copy_basic(qdisc_root_sleeping_running(sch), d, NULL, b);
 #endif
diff --git a/nss_qdisc/nss_qdisc.h b/nss_qdisc/nss_qdisc.h
index edb5da4..b344f83 100644
--- a/nss_qdisc/nss_qdisc.h
+++ b/nss_qdisc/nss_qdisc.h
@@ -217,7 +217,11 @@ struct nss_qdisc {
 						/* Shaper configure callback for reading shaper specific
 						 * responses (e.g. memory size).
 						 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	struct gnet_stats_basic_packed bstats;	/* Basic class statistics */
+#else
+	struct gnet_stats_basic_sync bstats;	/* Basic class statistics */
+#endif
 	struct gnet_stats_queue qstats;		/* Qstats for use by classes */
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 13, 0))
 	atomic_t refcnt;			/* Reference count for class use */
@@ -464,7 +468,11 @@ extern int nss_qdisc_init(struct Qdisc *sch, struct nss_qdisc *nq, nss_shaper_no
  *  Wrapper around gnet_stats_copy_basic()
  */
 extern int nss_qdisc_gnet_stats_copy_basic(struct Qdisc *sch,
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 				struct gnet_dump *d, struct gnet_stats_basic_packed *b);
+#else
+				struct gnet_dump *d, struct gnet_stats_basic_sync *b);
+#endif
 
 /*
  * nss_qdisc_gnet_stats_copy_queue()
diff --git a/nss_qdisc/nss_qdisc_stats.c b/nss_qdisc/nss_qdisc_stats.c
index 4b7d791..7a4a8f4 100644
--- a/nss_qdisc/nss_qdisc_stats.c
+++ b/nss_qdisc/nss_qdisc_stats.c
@@ -160,7 +160,11 @@ static void nss_qdisc_stats_process_node_stats(struct nss_qdisc *nqr,
 {
 	struct Qdisc *qdisc;
 	struct nss_qdisc *nq;
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	struct gnet_stats_basic_packed *bstats;
+#else
+	struct gnet_stats_basic_sync *bstats;
+#endif
 	struct gnet_stats_queue *qstats;
 	uint32_t qos_tag = response->qos_tag;
 #if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 13, 0))
@@ -214,8 +218,13 @@ static void nss_qdisc_stats_process_node_stats(struct nss_qdisc *nqr,
 	 * Update qdisc->bstats
 	 */
 	spin_lock_bh(&nq->lock);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	bstats->bytes += (__u64)response->sn_stats.delta.dequeued_bytes;
 	bstats->packets += response->sn_stats.delta.dequeued_packets;
+#else
+	u64_stats_add(&bstats->bytes, (__u64)response->sn_stats.delta.dequeued_bytes);
+	u64_stats_add(&bstats->packets, response->sn_stats.delta.dequeued_packets);
+#endif
 
 	/*
 	 * Update qdisc->qstats
diff --git a/pvxlanmgr/nss_pvxlanmgr.c b/pvxlanmgr/nss_pvxlanmgr.c
index e1d749f..b3a0c42 100644
--- a/pvxlanmgr/nss_pvxlanmgr.c
+++ b/pvxlanmgr/nss_pvxlanmgr.c
@@ -177,7 +177,11 @@ static struct rtnl_link_stats64 *nss_pvxlanmgr_get_tunnel_stats(struct net_devic
 	 * Netdev seems to be incrementing rx_dropped because we don't give IP header.
 	 * So reset it as it's of no use for us.
 	 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	atomic_long_set(&dev->rx_dropped, 0);
+#else
+	atomic_long_set(&(dev)->stats.__rx_dropped, 0);
+#endif
 	priv = netdev_priv(dev);
 	memset(stats, 0, sizeof(struct rtnl_link_stats64));
 	memcpy(stats, &priv->stats, sizeof(struct rtnl_link_stats64));
@@ -305,7 +309,11 @@ static void nss_pvxlanmgr_dummy_netdev_setup(struct net_device *dev)
 	dev->priv_destructor = NULL;
 #endif
 
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	memcpy(dev->dev_addr, "\x00\x00\x00\x00\x00\x00", dev->addr_len);
+#else
+	memcpy((void *) dev->dev_addr, "\x00\x00\x00\x00\x00\x00", dev->addr_len);
+#endif
 	memset(dev->broadcast, 0xff, dev->addr_len);
 	memcpy(dev->perm_addr, dev->dev_addr, dev->addr_len);
 }
diff --git a/tls/nss_tlsmgr_tun.c b/tls/nss_tlsmgr_tun.c
index 5c5acc1..1f274c6 100644
--- a/tls/nss_tlsmgr_tun.c
+++ b/tls/nss_tlsmgr_tun.c
@@ -185,7 +185,11 @@ static void nss_tlsmgr_tun_setup(struct net_device *dev)
 	/*
 	 * Get the MAC address from the ethernet device
 	 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	random_ether_addr(dev->dev_addr);
+#else
+	eth_random_addr((u8 *) dev->dev_addr);
+#endif
 
 	memset(dev->broadcast, 0xff, dev->addr_len);
 	memcpy(dev->perm_addr, dev->dev_addr, dev->addr_len);
diff --git a/tunipip6/nss_connmgr_tunipip6.c b/tunipip6/nss_connmgr_tunipip6.c
index 838e0cd..1249dc3 100644
--- a/tunipip6/nss_connmgr_tunipip6.c
+++ b/tunipip6/nss_connmgr_tunipip6.c
@@ -354,11 +354,21 @@ static void nss_tunipip6_update_dev_stats(struct net_device *dev,
 
 	memset(&stats, 0, sizeof(stats));
 	if (interface_type == NSS_DYNAMIC_INTERFACE_TYPE_TUNIPIP6_INNER) {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 		stats.tx_packets = sync_stats->node_stats.tx_packets;
 		stats.tx_bytes = sync_stats->node_stats.tx_bytes;
+#else
+		u64_stats_set(&stats.tx_packets, sync_stats->node_stats.tx_packets);
+		u64_stats_set(&stats.tx_bytes, sync_stats->node_stats.tx_bytes);
+#endif
 	} else if (interface_type == NSS_DYNAMIC_INTERFACE_TYPE_TUNIPIP6_OUTER) {
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 		stats.rx_packets = sync_stats->node_stats.rx_packets;
 		stats.rx_bytes = sync_stats->node_stats.rx_bytes;
+#else
+		u64_stats_set(&stats.rx_packets, sync_stats->node_stats.rx_packets);
+		u64_stats_set(&stats.rx_bytes, sync_stats->node_stats.rx_bytes);
+#endif
 	} else {
 		nss_tunipip6_warning("%px: Invalid interface type received from NSS\n", dev);
 		return;
diff --git a/vlan/nss_vlan_mgr.c b/vlan/nss_vlan_mgr.c
index 5aadc54..639f41e 100644
--- a/vlan/nss_vlan_mgr.c
+++ b/vlan/nss_vlan_mgr.c
@@ -787,7 +787,7 @@ static struct nss_vlan_pvt *nss_vlan_mgr
 	}

 	v->mtu = dev->mtu;
-	ether_addr_copy(v->dev_addr, dev->dev_addr);
+	ether_addr_copy(v->dev_addr, (uint8_t *) dev->dev_addr);
 	v->ifindex = dev->ifindex;
 	v->refs = 1;

@@ -936,14 +936,14 @@ static int nss_vlan_mgr_changeaddr_event
 	}
 	spin_unlock(&vlan_mgr_ctx.lock);

-	if (nss_vlan_tx_set_mac_addr_msg(v_pvt->nss_if, dev->dev_addr) != NSS_TX_SUCCESS) {
+	if (nss_vlan_tx_set_mac_addr_msg(v_pvt->nss_if, (uint8_t *) dev->dev_addr) != NSS_TX_SUCCESS) {
 		nss_vlan_mgr_warn("%s: Failed to send change MAC address message to NSS\n", dev->name);
 		nss_vlan_mgr_instance_deref(v_pvt);
 		return NOTIFY_BAD;
 	}

 	spin_lock(&vlan_mgr_ctx.lock);
-	ether_addr_copy(v_pvt->dev_addr, dev->dev_addr);
+	ether_addr_copy(v_pvt->dev_addr, (uint8_t *) dev->dev_addr);
 	spin_unlock(&vlan_mgr_ctx.lock);
 	nss_vlan_mgr_trace("%s: MAC changed to %pM, updated NSS\n", dev->name, dev->dev_addr);
 	nss_vlan_mgr_instance_deref(v_pvt);

diff --git a/vxlanmgr/nss_vxlanmgr_tunnel.c b/vxlanmgr/nss_vxlanmgr_tunnel.c
index 44df78a..9af697c 100644
--- a/vxlanmgr/nss_vxlanmgr_tunnel.c
+++ b/vxlanmgr/nss_vxlanmgr_tunnel.c
@@ -489,8 +489,13 @@ static void nss_vxlanmgr_tunnel_inner_stats(struct nss_vxlanmgr_tun_ctx *tun_ctx
 
 	tstats = this_cpu_ptr(dev->tstats);
 	u64_stats_update_begin(&tstats->syncp);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	tstats->tx_packets += stats->node_stats.tx_packets;
 	tstats->tx_bytes += stats->node_stats.tx_bytes;
+#else
+	u64_stats_add(&tstats->tx_packets, stats->node_stats.tx_packets);
+	u64_stats_add(&tstats->tx_bytes, stats->node_stats.tx_bytes);
+#endif
 	u64_stats_update_end(&tstats->syncp);
 	netdev_stats->tx_dropped += dropped;
 }
@@ -526,8 +531,13 @@ static void nss_vxlanmgr_tunnel_outer_stats(struct nss_vxlanmgr_tun_ctx *tun_ctx
 
 	tstats = this_cpu_ptr(dev->tstats);
 	u64_stats_update_begin(&tstats->syncp);
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	tstats->rx_packets += stats->node_stats.tx_packets;
 	tstats->rx_bytes += stats->node_stats.tx_bytes;
+#else
+	u64_stats_add(&tstats->rx_packets, stats->node_stats.tx_packets);
+	u64_stats_add(&tstats->rx_bytes, stats->node_stats.tx_bytes);
+#endif
 	u64_stats_update_end(&tstats->syncp);
 	netdev_stats->rx_dropped += dropped;
 	dev_put(dev);
diff --git a/ipsecmgr/v1.0/nss_ipsecmgr.c b/ipsecmgr/v1.0/nss_ipsecmgr.c
index 64d04fa..0f13f7a 100644
--- a/ipsecmgr/v1.0/nss_ipsecmgr.c
+++ b/ipsecmgr/v1.0/nss_ipsecmgr.c
@@ -445,7 +445,11 @@ static void nss_ipsecmgr_tunnel_setup(st
 	/*
 	 * get the MAC address from the ethernet device
 	 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	random_ether_addr(dev->dev_addr);
+#else
+	eth_random_addr((u8 *) dev->dev_addr);
+#endif
 
 	memset(dev->broadcast, 0xff, dev->addr_len);
 	memcpy(dev->perm_addr, dev->dev_addr, dev->addr_len);
diff --git a/ipsecmgr/v2.0/nss_ipsecmgr_tunnel.c b/ipsecmgr/v2.0/nss_ipsecmgr_tunnel.c
index 78c6376..959c7df 100644
--- a/ipsecmgr/v2.0/nss_ipsecmgr_tunnel.c
+++ b/ipsecmgr/v2.0/nss_ipsecmgr_tunnel.c
@@ -445,7 +445,11 @@ static void nss_ipsecmgr_tunnel_setup(st
 	/*
 	 * Get the MAC address from the ethernet device
 	 */
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(5, 16, 0))
 	random_ether_addr(dev->dev_addr);
+#else
+	eth_random_addr((u8 *) dev->dev_addr);
+#endif
 
 	memset(dev->broadcast, 0xff, dev->addr_len);
 	memcpy(dev->perm_addr, dev->dev_addr, dev->addr_len);
