diff --git a/image/mt7621.mk b/image/mt7621.mk
index 28ae0d4..0c64081 100644
--- a/image/mt7621.mk
+++ b/image/mt7621.mk
@@ -257,6 +257,7 @@ define Device/k2p
   DTS := K2P
   IMAGE_SIZE := $(ralink_default_fw_size_16M)
   DEVICE_TITLE := Phicomm K2P
+  DEVICE_PACKAGES := luci-app-mtwifi
 endef
 TARGET_DEVICES += k2p
 
@@ -272,7 +273,7 @@ define Device/xiaomi_mir3p
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
   IMAGE/factory.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | check-size $$$$(IMAGE_SIZE)
   DEVICE_PACKAGES := \
-	kmod-mt7615e kmod-usb3 kmod-usb-ledtrig-usbport wpad-basic \
+	luci-app-mtwifi kmod-usb3 kmod-usb-ledtrig-usbport \
 	uboot-envtools
 endef
 TARGET_DEVICES += xiaomi_mir3p
