diff --git a/realtek.mk b/realtek.mk
index e06c3a6..e939910 100644
--- a/realtek.mk
+++ b/realtek.mk
@@ -101,6 +101,13 @@ define Package/rtl8822be-firmware/install
 endef
 $(eval $(call BuildPackage,rtl8822be-firmware))
 
+Package/rtl8821cu-firmware = $(call Package/firmware-default,RealTek RTL8821CU firmware)
+define Package/rtl8821cu-firmware/install
+	$(INSTALL_DIR) $(1)/lib/firmware/rtw88
+	$(INSTALL_DATA) $(PKG_BUILD_DIR)/rtw88/rtw8821c_fw.bin $(1)/lib/firmware/rtw88
+endef
+$(eval $(call BuildPackage,rtl8821cu-firmware))
+
 Package/rtl8822ce-firmware = $(call Package/firmware-default,RealTek RTL8822CE firmware)
 define Package/rtl8822ce-firmware/install
 	$(INSTALL_DIR) $(1)/lib/firmware/rtw88
