--- a/drivers/net/bonding/bond_main.c
+++ b/drivers/net/bonding/bond_main.c
@@ -246,6 +246,9 @@ static const struct flow_dissector_key flow_keys_bonding_keys[] = {
 	},
 };
 
+/* QCA NSS bonding support */
+static unsigned long bond_id_mask = 0xFFFFFFF0;
+
 static struct flow_dissector flow_keys_bonding __read_mostly;
 
 /*-------------------------- Forward declarations ---------------------------*/
@@ -3861,6 +3864,23 @@ static int bond_get_lowest_level_rcu(struct net_device *dev)
 }
 #endif
 
+/* QCA NSS bonding support */
+int bond_get_id(struct net_device *bond_dev)
+{
+    struct bonding *bond;
+    int bond_id = 0;
+
+    if (!((bond_dev->priv_flags & IFF_BONDING) &&
+          (bond_dev->flags & IFF_MASTER)))
+        return -EINVAL;
+
+    bond = netdev_priv(bond_dev);
+    bond_id = bond->id;
+
+    return bond_id;
+}
+EXPORT_SYMBOL(bond_get_id);
+
 static void bond_get_stats(struct net_device *bond_dev,
 			   struct rtnl_link_stats64 *stats)
 {
@@ -4835,6 +4855,10 @@ static void bond_destructor(struct net_device *bond_dev)
 	struct bonding *bond = netdev_priv(bond_dev);
 	if (bond->wq)
 		destroy_workqueue(bond->wq);
+
+    /* QCA NSS bonding support */
+    if (bond->id != (~0U))
+        clear_bit(bond->id, &bond_id_mask);
 }
 
 void bond_setup(struct net_device *bond_dev)
@@ -5402,7 +5426,14 @@ int bond_create(struct net *net, const char *name)
 
 	bond_work_init_all(bond);
 
-	rtnl_unlock();
+    /* QCA NSS bonding support */
+    bond->id = ~0U;
+    if (bond_id_mask != (~0UL)) {
+        bond->id = (u32)ffz(bond_id_mask);
+        set_bit(bond->id, &bond_id_mask);    
+    }
+
+    rtnl_unlock();
 	return 0;
 }
 
--- a/include/net/bonding.h
+++ b/include/net/bonding.h
@@ -256,6 +256,7 @@ struct bonding {
 	/* protecting ipsec_list */
 	spinlock_t ipsec_lock;
 #endif /* CONFIG_XFRM_OFFLOAD */
+    u32    id; /* QCA NSS bonding */
 };
 
 #define bond_slave_get_rcu(dev) \
@@ -628,6 +629,7 @@ struct bond_net {
 
 int bond_arp_rcv(const struct sk_buff *skb, struct bonding *bond, struct slave *slave);
 netdev_tx_t bond_dev_queue_xmit(struct bonding *bond, struct sk_buff *skb, struct net_device *slave_dev);
+int bond_get_id(struct net_device *bond_dev); /* QCA NSS bonding support */
 int bond_create(struct net *net, const char *name);
 int bond_create_sysfs(struct bond_net *net);
 void bond_destroy_sysfs(struct bond_net *net);
