diff --git a/luasrc/sys.lua b/luasrc/sys.lua
index aa00766..5b58c84 100644
--- a/luasrc/sys.lua
+++ b/luasrc/sys.lua
@@ -525,7 +525,7 @@ function user.checkpasswd(username, pass)
 end
 
 function user.setpasswd(username, password)
-	return os.execute("(echo %s; sleep 1; echo %s) | passwd %s >/dev/null 2>&1" %{
+	return os.execute("(echo %s; sleep 1; echo %s) | busybox passwd %s >/dev/null 2>&1" %{
 		luci.util.shellquote(password),
 		luci.util.shellquote(password),
 		luci.util.shellquote(username)
diff --git a/root/usr/libexec/rpcd/luci b/root/usr/libexec/rpcd/luci
index 17fb391..1e57426 100755
--- a/root/usr/libexec/rpcd/luci
+++ b/root/usr/libexec/rpcd/luci
@@ -443,7 +443,7 @@ local methods = {
 		call = function(args)
 			local util = require "luci.util"
 			return {
-				result = (os.execute("(echo %s; sleep 1; echo %s) | passwd %s >/dev/null 2>&1" %{
+				result = (os.execute("(echo %s; sleep 1; echo %s) | busybox passwd %s >/dev/null 2>&1" %{
 					luci.util.shellquote(args.password),
 					luci.util.shellquote(args.password),
 					luci.util.shellquote(args.username)
@@ -614,6 +614,37 @@ local methods = {
 			end
 			return { result = res }
 		end
+	},
+
+	getCPUBench = {
+		call = function()
+			local sys = require "luci.sys"
+			local cpubench = {}
+
+			cpubench.cpubench = sys.exec("cat /etc/bench.log 2>/dev/null")
+			return cpubench
+		end
+	},
+
+	getCPUUsage = {
+		call = function()
+			local sys = require "luci.sys"
+			local cpuusage = {}
+
+			cpuusage.cpuusage = sys.exec("top -n1 | awk '/^CPU/ {printf(\"%d%\", 100 - $8)}'") or "6%"
+			return cpuusage
+		end
+	},
+
+	getOnlineUsers = {
+		call = function()
+			local sys = require "luci.sys"
+			local onlineusers = {}
+
+			onlineusers.onlineusers = sys.exec("cat /proc/net/arp | grep 'br-lan' | grep '0x2' | wc -l")
+			return onlineusers
+
+		end
 	}
 }
 
