diff -uNr a/Config.in b/Config.in
--- a/Config.in	2026-01-03 14:12:16.942136000 +0800
+++ b/Config.in	2026-01-03 14:45:23.402076267 +0800
@@ -4,7 +4,7 @@
 
 config OPENSSL_OPTIMIZE_SPEED
 	bool
-	default y if x86_64 || i386
+	default y
 	prompt "Enable optimization for speed instead of size"
 	select OPENSSL_WITH_ASM
 	help
@@ -34,6 +34,15 @@
 		This will enable kTLS support, allowing data encryption
 		operations to be performed in kernel space.
 
+config OPENSSL_QUIC
+	bool
+	prompt "Enable QUIC support"
+	select OPENSSL_WITH_TLS13
+	help
+		Enable support for QUIC (Quick UDP Internet Connections) in OpenSSL 3.
+		This allows OpenSSL to provide cryptographic support for QUIC-based protocols,
+		such as HTTP/3. QUIC requires TLSv1.3 to be enabled.
+
 config OPENSSL_WITH_ASM
 	bool
 	default y
@@ -46,9 +55,9 @@
 
 config OPENSSL_WITH_SSE2
 	bool
-	default y if !TARGET_x86_legacy && !TARGET_x86_geode
+	default y if x86_64 || i386 && !TARGET_x86_legacy
 	prompt "Enable use of x86 SSE2 instructions"
-	depends on OPENSSL_WITH_ASM && i386
+	depends on OPENSSL_WITH_ASM && x86_64 || i386
 	help
 		Use of SSE2 instructions greatly increase performance with a
 		minimum increase in package size, but it will bring no benefit
@@ -141,7 +150,7 @@
 
 config OPENSSL_PREFER_CHACHA_OVER_GCM
 	bool
-	default y if !x86_64 && !aarch64
+	default y if !x86_64
 	prompt "Prefer ChaCha20-Poly1305 over AES-GCM by default"
 	depends on OPENSSL_WITH_CHACHA_POLY1305
 	help
@@ -269,6 +278,7 @@
 
 config OPENSSL_ENGINE_BUILTIN
 	bool "Build chosen engines into libcrypto"
+	default y
 	depends on OPENSSL_ENGINE
 	help
 		This builds all chosen engines into libcrypto.so, instead of building
@@ -278,6 +288,7 @@
 
 config OPENSSL_ENGINE_BUILTIN_AFALG
 	bool
+	default y
 	prompt "Acceleration support through AF_ALG sockets engine"
 	depends on OPENSSL_ENGINE_BUILTIN && KERNEL_AIO
 	select PACKAGE_libopenssl-conf
@@ -287,6 +298,7 @@
 
 config OPENSSL_ENGINE_BUILTIN_DEVCRYPTO
 	bool
+	default y
 	prompt "Acceleration support through /dev/crypto"
 	depends on OPENSSL_ENGINE_BUILTIN
 	select PACKAGE_libopenssl-conf
diff -uNr a/Makefile b/Makefile
--- a/Makefile	2026-01-03 14:12:16.942136000 +0800
+++ b/Makefile	2026-01-03 14:38:14.838089258 +0800
@@ -38,6 +38,7 @@
 	CONFIG_OPENSSL_PREFER_CHACHA_OVER_GCM \
 	CONFIG_OPENSSL_SMALL_FOOTPRINT \
 	CONFIG_OPENSSL_KTLS \
+	CONFIG_OPENSSL_QUIC \
 	CONFIG_OPENSSL_WITH_ARIA \
 	CONFIG_OPENSSL_WITH_ASM \
 	CONFIG_OPENSSL_WITH_ASYNC \
@@ -298,6 +299,10 @@
   OPENSSL_OPTIONS += enable-ktls
 endif
 
+ifdef CONFIG_OPENSSL_QUIC
+  OPENSSL_OPTIONS += enable-quic
+endif
+
 ifdef CONFIG_OPENSSL_ENGINE
   ifdef CONFIG_OPENSSL_ENGINE_BUILTIN
     OPENSSL_OPTIONS += disable-dynamic-engine
