diff --git a/mt7621/base-files/etc/board.d/02_network b/mt7621/base-files/etc/board.d/02_network
index 03cc921..0dd1a55 100755
--- a/mt7621/base-files/etc/board.d/02_network
+++ b/mt7621/base-files/etc/board.d/02_network
@@ -44,6 +44,9 @@ ramips_setup_interfaces()
 	mikrotik,routerboard-750gr3)
 		ucidef_set_interfaces_lan_wan "lan2 lan3 lan4 lan5" "wan"
 		;;
+	phicomm,k2p)
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4 ra0 rax0" "wan"
+		;;
 	ubnt,edgerouter-x)
 		ucidef_set_interface_lan_wan "eth1 eth2 eth3 eth4" "eth0"
 		;;
