diff --git a/files/adblock.init b/files/adblock.init
index 732c8a6..52c35a0 100755
--- a/files/adblock.init
+++ b/files/adblock.init
@@ -44,6 +44,7 @@ start_service() {
 
 reload_service() {
 	rc_procd start_service reload
+	/etc/init.d/adblock report >/dev/null 2>&1
 }
 
 stop_service() {
@@ -52,6 +53,7 @@ stop_service() {
 
 restart() {
 	rc_procd start_service restart
+	/etc/init.d/adblock report >/dev/null 2>&1
 }
 
 suspend() {
