--- a/files/etc/init.d/zerotier
+++ b/files/etc/init.d/zerotier
@@ -18,7 +18,7 @@ start_instance() {
 	local args=""
 
 	if ! section_enabled "$cfg"; then
-		echo "disabled in config"
+		echo "disabled in /etc/config/zerotier"
 		return 1
 	fi
 
@@ -35,9 +35,10 @@ start_instance() {
 
 	# Create link or copy files from CONFIG_PATH to config_path
 	if [ -n "$config_path" -a "$config_path" != "$path" ]; then
+		# Create the config path to init and persist
 		if [ ! -d "$config_path" ]; then
-			echo "ZeroTier config_path does not exist: $config_path" 1>&2
-			return
+			echo "ZeroTier config_path does not exist: $config_path, create..."
+			mkdir -p $config_path
 		fi
 
 		# ensure that the target exists
@@ -60,7 +61,7 @@ start_instance() {
 		args="$args -p${port}"
 	fi
 
-	if [ -z "$secret" ]; then
+	if [ -z "$secret" -a ! -f $path/identity.secret ]; then
 		echo "Generate secret - please wait..."
 		local sf="/tmp/zt.$cfg.secret"
 
@@ -106,6 +107,9 @@ start_service() {
 stop_instance() {
 	local cfg="$1"
 
+	# Remove existing networks
+	rm -f ${CONFIG_PATH}_${cfg}/networks.d/*.conf
+
 	# Remove existing link or folder
 	rm -rf ${CONFIG_PATH}_${cfg}
 }
