diff --git a/htdocs/luci-static/resources/view/aria2/config.js b/htdocs/luci-static/resources/view/aria2/config.js
index 84b5075..01e3b0e 100644
--- a/htdocs/luci-static/resources/view/aria2/config.js
+++ b/htdocs/luci-static/resources/view/aria2/config.js
@@ -128,7 +128,7 @@ function getWebFrontInstalled() {
 function openWebInterface(path) {
 	var host = window.location.host;
 	var protocol = window.location.protocol;
-	window.open(protocol + '//' + host + '/' + path);
+	window.open(protocol + '//' + host + '/' + path + '/');
 };
 
 function showRPCURL(section_id, useWS, inputEl) {
