diff --git a/Makefile b/Makefile
index a80eccd..502eb83 100644
--- a/Makefile
+++ b/Makefile
@@ -53,6 +53,7 @@ define Package/ntfs-3g/description
   Contains:
    - ntfs-3g
    - ntfs-3g.probe
+   - mount.ntfs (symlink to ntfs-3g)
    - mount.ntfs-3g (symlink to ntfs-3g)
 
 endef
@@ -160,6 +161,7 @@ define Package/ntfs-3g/install
 	$(INSTALL_DIR) $(1)/usr/lib
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libntfs-3g.so.* $(1)/usr/lib/
 	$(INSTALL_DIR) $(1)/sbin
+	$(LN) ../usr/bin/ntfs-3g $(1)/sbin/mount.ntfs
 	$(LN) ../usr/bin/ntfs-3g $(1)/sbin/mount.ntfs-3g
 endef
 
