--- a/files/lib/netifd/wireless/mac80211.sh
+++ b/files/lib/netifd/wireless/mac80211.sh
@@ -280,22 +280,38 @@ mac80211_hostapd_setup_base() {
 			vht_center_seg0=$idx
 		;;
 	esac
-	[ "$band" = "5g" ] && {
-		json_get_vars background_radar:0
-
-		[ "$background_radar" -eq 1 ] && append base_cfg "enable_background_radar=1" "$N"
-	}
-	[ "$band" = "6g" ] && {
-		op_class=
+	if [ "$band" = "2g" ]; then
 		case "$htmode" in
-			HE20) op_class=131;;
-			HE*) op_class=$((132 + $vht_oper_chwidth))
+			VHT20|VHT40)
+				vendor_vht=1
+			;;
+			HT20|HT40)
+				vendor_vht=0
+			;;
+			*)
+				htmode="HT20"
+				vendor_vht=0
+			;;
 		esac
-		[ -n "$op_class" ] && append base_cfg "op_class=$op_class" "$N"
-	}
-	[ "$hwmode" = "a" ] || enable_ac=0
+	else
+		[ "$band" = "5g" ] && {
+			json_get_vars background_radar:0
+
+			[ "$background_radar" -eq 1 ] && append base_cfg "enable_background_radar=1" "$N"
+		}
+		[ "$band" = "6g" ] && {
+			op_class=
+			case "$htmode" in
+				HE20) op_class=131;;
+				HE*) op_class=$((132 + $vht_oper_chwidth))
+			esac
+			[ -n "$op_class" ] && append base_cfg "op_class=$op_class" "$N"
+		}
+
+		vendor_vht=0
+	fi
 
-	if [ "$enable_ac" != "0" ]; then
+	if [ "$enable_ac" != "0" -o "$vendor_vht" = "1" ]; then
 		json_get_vars \
 			rxldpc:1 \
 			short_gi_80:1 \
--- a/files/lib/wifi/mac80211.sh
+++ b/files/lib/wifi/mac80211.sh
@@ -203,14 +203,16 @@ detect_mac80211() {
 			set wireless.${name}.channel=${channel}
 			set wireless.${name}.band=${mode_band}
 			set wireless.${name}.htmode=$htmode
-			set wireless.${name}.disabled=1
+			set wireless.${name}.disabled=0
+			set wireless.${name}.country=US
 
 			set wireless.default_${name}=wifi-iface
 			set wireless.default_${name}.device=${name}
 			set wireless.default_${name}.network=lan
 			set wireless.default_${name}.mode=ap
 			set wireless.default_${name}.ssid=OpenWrt
-			set wireless.default_${name}.encryption=none
+			set wireless.default_${name}.encryption=psk2
+			set wireless.default_${name}.key=1234567890
 EOF
 		uci -q commit wireless
 	done
