diff --git a/Makefile b/Makefile
index 32d554f..92dcbdf 100644
--- a/Makefile
+++ b/Makefile
@@ -19,8 +19,7 @@ define Package/transmission-web-control
   SECTION:=net
   CATEGORY:=Network
   SUBMENU:=BitTorrent
-  DEPENDS:=@(PACKAGE_transmission-daemon-openssl||PACKAGE_transmission-daemon-mbedtls)
-  CONFLICTS:=transmission-web
+  DEPENDS:=+transmission-daemon-openssl
   TITLE:=Transmission Web Control
   URL:=https://github.com/ronggang/transmission-web-control
   PKGARCH:=all
