diff --git a/files/sysctl-tcp-bbrplus.conf b/files/sysctl-tcp-bbrplus.conf
new file mode 100644
index 0000000..e1c3cd2
--- /dev/null
+++ b/files/sysctl-tcp-bbrplus.conf
@@ -0,0 +1,5 @@
+# Do not edit, changes to this file will be lost on upgrades
+# /etc/sysctl.conf can be used to customize sysctl settings
+
+net.core.default_qdisc=fq_codel
+net.ipv4.tcp_congestion_control=bbrplus
diff --git a/files/sysctl-tcp-nanqinlang.conf b/files/sysctl-tcp-nanqinlang.conf
new file mode 100644
index 0000000..107dd37
--- /dev/null
+++ b/files/sysctl-tcp-nanqinlang.conf
@@ -0,0 +1,5 @@
+# Do not edit, changes to this file will be lost on upgrades
+# /etc/sysctl.conf can be used to customize sysctl settings
+
+net.core.default_qdisc=fq_codel
+net.ipv4.tcp_congestion_control=nanqinlang
diff --git a/files/sysctl-tcp-tsunami.conf b/files/sysctl-tcp-tsunami.conf
new file mode 100644
index 0000000..58932aa
--- /dev/null
+++ b/files/sysctl-tcp-tsunami.conf
@@ -0,0 +1,5 @@
+# Do not edit, changes to this file will be lost on upgrades
+# /etc/sysctl.conf can be used to customize sysctl settings
+
+net.core.default_qdisc=fq_codel
+net.ipv4.tcp_congestion_control=tsunami
diff --git a/files/sysctl-tcp-tsunamio.conf b/files/sysctl-tcp-tsunamio.conf
new file mode 100644
index 0000000..b0a793d
--- /dev/null
+++ b/files/sysctl-tcp-tsunamio.conf
@@ -0,0 +1,5 @@
+# Do not edit, changes to this file will be lost on upgrades
+# /etc/sysctl.conf can be used to customize sysctl settings
+
+net.core.default_qdisc=fq_codel
+net.ipv4.tcp_congestion_control=tsunamio
diff --git a/modules/netsupport.mk b/modules/netsupport.mk
index 76187df..68f9e9b 100644
--- a/modules/netsupport.mk
+++ b/modules/netsupport.mk
@@ -968,6 +968,123 @@ endef
 $(eval $(call KernelPackage,tcp-bbr))
 
 
+define KernelPackage/tcp-nanqinlang
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=NANQINLANG TCP congestion control
+  KCONFIG:= \
+            CONFIG_TCP_CONG_NANQINLANG \
+            CONFIG_NET_SCH_DEFAULT_BBRP=y \
+            CONFIG_NET_SCH_DEFAULT=y \
+            CONFIG_DEFAULT_FQ_CODEL=y \
+            CONFIG_DEFAULT_NET_SCH="fq_codel"
+  FILES:=$(LINUX_DIR)/net/ipv4/tcp_nanqinlang.ko
+  AUTOLOAD:=$(call AutoProbe,tcp_nanqinlang)
+endef
+
+define KernelPackage/tcp-nanqinlang/description
+ NANQINLANG is an enhanced version of BBR (Bottleneck Bandwidth and RTT).
+ Originally introduced by dog250 & cx9208.
+ Same as BBR, requires the fq ("Fair Queue") pacing packet scheduler.
+endef
+
+TCP_NANQINLANG_SYSCTL_CONF:=sysctl-tcp-nanqinlang.conf
+
+define KernelPackage/tcp-nanqinlang/install
+	$(INSTALL_DIR) $(1)/etc/sysctl.d
+	$(INSTALL_DATA) ./files/$(TCP_NANQINLANG_SYSCTL_CONF) $(1)/etc/sysctl.d/12-tcp-nanqinlang.conf
+endef
+
+$(eval $(call KernelPackage,tcp-nanqinlang))
+
+
+define KernelPackage/tcp-tsunami
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=TSUNAMI TCP congestion control
+  KCONFIG:= \
+            CONFIG_TCP_CONG_TSUNAMI \
+            CONFIG_NET_SCH_DEFAULT_BBRP=y \
+            CONFIG_NET_SCH_DEFAULT=y \
+            CONFIG_DEFAULT_FQ_CODEL=y \
+            CONFIG_DEFAULT_NET_SCH="fq_codel"
+  FILES:=$(LINUX_DIR)/net/ipv4/tcp_tsunami.ko
+  AUTOLOAD:=$(call AutoProbe,tcp_tsunami)
+endef
+
+define KernelPackage/tcp-tsunami/description
+ TSUNAMI is an enhanced version of BBR (Bottleneck Bandwidth and RTT).
+ Originally introduced by dog250 & cx9208.
+ Same as BBR, requires the fq ("Fair Queue") pacing packet scheduler.
+endef
+
+TCP_TSUNAMI_SYSCTL_CONF:=sysctl-tcp-tsunami.conf
+
+define KernelPackage/tcp-tsunami/install
+	$(INSTALL_DIR) $(1)/etc/sysctl.d
+	$(INSTALL_DATA) ./files/$(TCP_TSUNAMI_SYSCTL_CONF) $(1)/etc/sysctl.d/12-tcp-tsunami.conf
+endef
+
+$(eval $(call KernelPackage,tcp-tsunami))
+
+
+define KernelPackage/tcp-tsunamio
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+#  DEPENDS:=@LINUX_5_4
+  TITLE:=TSUNAMIO TCP congestion control
+  KCONFIG:= \
+            CONFIG_TCP_CONG_TSUNAMIO \
+            CONFIG_NET_SCH_DEFAULT_BBRP=y \
+            CONFIG_NET_SCH_DEFAULT=y \
+            CONFIG_DEFAULT_FQ_CODEL=y \
+            CONFIG_DEFAULT_NET_SCH="fq_codel"
+  FILES:=$(LINUX_DIR)/net/ipv4/tcp_tsunamio.ko
+  AUTOLOAD:=$(call AutoProbe,tcp_tsunamio)
+endef
+
+define KernelPackage/tcp-tsunamio/description
+ TSUNAMIO is an enhanced version of BBR (Bottleneck Bandwidth and RTT).
+ Originally introduced by dog250 & cx9208.
+ Same as BBR, requires the fq ("Fair Queue") pacing packet scheduler.
+endef
+
+TCP_TSUNAMIO_SYSCTL_CONF:=sysctl-tcp-tsunamio.conf
+
+define KernelPackage/tcp-tsunamio/install
+	$(INSTALL_DIR) $(1)/etc/sysctl.d
+	$(INSTALL_DATA) ./files/$(TCP_TSUNAMIO_SYSCTL_CONF) $(1)/etc/sysctl.d/12-tcp-tsunamio.conf
+endef
+
+$(eval $(call KernelPackage,tcp-tsunamio))
+
+
+define KernelPackage/tcp-bbrplus
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=BBRPLUS TCP congestion control
+  KCONFIG:= \
+            CONFIG_TCP_CONG_BBRPLUS \
+            CONFIG_NET_SCH_DEFAULT_BBRP=y \
+            CONFIG_NET_SCH_DEFAULT=y \
+            CONFIG_DEFAULT_FQ_CODEL=y \
+            CONFIG_DEFAULT_NET_SCH="fq_codel"
+  FILES:=$(LINUX_DIR)/net/ipv4/tcp_bbrplus.ko
+  AUTOLOAD:=$(call AutoProbe,tcp_bbrplus)
+endef
+
+define KernelPackage/tcp-bbrplus/description
+ BBRplus is an enhanced version of BBR (Bottleneck Bandwidth and RTT).
+ Originally introduced by dog250 & cx9208.
+ Same as BBR, requires the fq ("Fair Queue") pacing packet scheduler.
+endef
+
+TCP_BBRPLUS_SYSCTL_CONF:=sysctl-tcp-bbrplus.conf
+
+define KernelPackage/tcp-bbrplus/install
+	$(INSTALL_DIR) $(1)/etc/sysctl.d
+	$(INSTALL_DATA) ./files/$(TCP_BBRPLUS_SYSCTL_CONF) $(1)/etc/sysctl.d/12-tcp-bbrplus.conf
+endef
+
+$(eval $(call KernelPackage,tcp-bbrplus))
+
+
 define KernelPackage/tcp-hybla
   SUBMENU:=$(NETWORK_SUPPORT_MENU)
   TITLE:=TCP-Hybla congestion control algorithm
@@ -1362,3 +1479,67 @@ define KernelPackage/netconsole/description
 endef
 
 $(eval $(call KernelPackage,netconsole))
+
+
+define KernelPackage/qrtr
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=Qualcomm IPC Router support
+  HIDDEN:=1
+  DEPENDS:=@LINUX_5_15
+  KCONFIG:=CONFIG_QRTR
+  FILES:= \
+  $(LINUX_DIR)/net/qrtr/qrtr.ko \
+  $(LINUX_DIR)/net/qrtr/ns.ko
+  AUTOLOAD:=$(call AutoProbe,qrtr)
+endef
+
+define KernelPackage/qrtr/description
+ Qualcomm IPC Router support
+endef
+
+$(eval $(call KernelPackage,qrtr))
+
+define KernelPackage/qrtr-tun
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=TUN device for Qualcomm IPC Router
+  DEPENDS:=+kmod-qrtr
+  KCONFIG:=CONFIG_QRTR_TUN
+  FILES:= $(LINUX_DIR)/net/qrtr/qrtr-tun.ko
+  AUTOLOAD:=$(call AutoProbe,qrtr-tun)
+endef
+
+define KernelPackage/qrtr-tun/description
+ TUN device for Qualcomm IPC Router
+endef
+
+$(eval $(call KernelPackage,qrtr-tun))
+
+define KernelPackage/qrtr-smd
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=SMD IPC Router channels
+  DEPENDS:=+kmod-qrtr @TARGET_ipq807x
+  KCONFIG:=CONFIG_QRTR_SMD
+  FILES:= $(LINUX_DIR)/net/qrtr/qrtr-smd.ko
+  AUTOLOAD:=$(call AutoProbe,qrtr-smd)
+endef
+
+define KernelPackage/qrtr-smd/description
+ SMD IPC Router channels
+endef
+
+$(eval $(call KernelPackage,qrtr-smd))
+
+define KernelPackage/qrtr-mhi
+  SUBMENU:=$(NETWORK_SUPPORT_MENU)
+  TITLE:=MHI IPC Router channels
+  DEPENDS:=+kmod-mhi-bus +kmod-qrtr
+  KCONFIG:=CONFIG_QRTR_MHI
+  FILES:= $(LINUX_DIR)/net/qrtr/qrtr-mhi.ko
+  AUTOLOAD:=$(call AutoProbe,qrtr-mhi)
+endef
+
+define KernelPackage/qrtr-mhi/description
+ MHI IPC Router channels
+endef
+
+$(eval $(call KernelPackage,qrtr-mhi))
