--- a/files/sysctl-nf-conntrack.conf
+++ b/files/sysctl-nf-conntrack.conf
@@ -6,3 +6,6 @@ net.netfilter.nf_conntrack_checksum=0
 net.netfilter.nf_conntrack_tcp_timeout_established=7440
 net.netfilter.nf_conntrack_udp_timeout=60
 net.netfilter.nf_conntrack_udp_timeout_stream=180
+net.netfilter.nf_conntrack_helper=1
+net.netfilter.nf_conntrack_buckets=16384
+net.netfilter.nf_conntrack_expect_max=16384
