--- a/Config_ssl.in
+++ b/Config_ssl.in
@@ -264,4 +264,11 @@ config NGINX_TS_MODULE
 		Add support for MPEG-TS Live Module module.
 	default n
 
+config NGINX_NJS_MODULE
+	bool
+	prompt "Enable NJS module"
+	help
+		Add support for Javascript dynamic module.
+	default n
+
 endmenu
--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,8 @@ PKG_SOURCE_URL:=https://nginx.org/download/
 PKG_HASH:=14774aae0d151da350417efc4afda5cce5035056e71894836797e1f6e2d1175a
 
 PKG_MAINTAINER:=Thomas Heil <heil@terminal-consulting.de> \
-				Ansuel Smith <ansuelsmth@gmail.com>
+				Ansuel Smith <ansuelsmth@gmail.com> \
+				Nicolas Penin <nicolas.penin@dragon-angel.fr>
 PKG_LICENSE:=2-clause BSD-like license
 PKG_CPE_ID:=cpe:/a:nginx:nginx
 
@@ -71,6 +72,7 @@ PKG_CONFIG_DEPENDS := \
 	CONFIG_NGINX_STREAM_SSL_PREREAD_MODULE \
 	CONFIG_NGINX_RTMP_MODULE \
 	CONFIG_NGINX_TS_MODULE \
+	CONFIG_NGINX_NJS_MODULE \
 	CONFIG_OPENSSL_ENGINE \
 	CONFIG_OPENSSL_WITH_NPN
 
@@ -288,6 +290,9 @@ ifneq ($(BUILD_VARIANT),all-module)
   ifeq ($(CONFIG_NGINX_TS_MODULE),y)
     ADDITIONAL_MODULES += --add-module=$(PKG_BUILD_DIR)/nginx-ts
   endif
+  ifeq ($(CONFIG_NGINX_NJS_MODULE),y)
+    ADDITIONAL_MODULES += --add-module=$(PKG_BUILD_DIR)/nginx-njs
+  endif
 else
   CONFIG_NGINX_HEADERS_MORE:=y
   CONFIG_NGINX_HTTP_BROTLI:=y
@@ -327,6 +332,21 @@ define Package/nginx-mod-luci/description
  Support file for LuCI in nginx. Include custom nginx configuration, autostart script for uwsgi.
 endef
 
+define Package/nginx-mod-njs
+  TITLE:=njs in nginx
+  SECTION:=net
+  CATEGORY:=Network
+  SUBMENU:=Web Servers/Proxies
+  TITLE:=dynamic module for Nginx
+  URL:=http://nginx.org/
+  DEPENDS:=+nginx
+  CONFLICTS:=nginx-all-module
+endef
+
+define Package/nginx-mod-njs/description
+ Dynamic njs module for Nginx.
+endef
+
 
 TARGET_CFLAGS += -fvisibility=hidden -ffunction-sections -fdata-sections -DNGX_LUA_NO_BY_LUA_BLOCK
 TARGET_LDFLAGS += -Wl,--gc-sections
@@ -449,6 +469,20 @@ define Prepare/nginx-ts
 	xzcat $(DL_DIR)/$(FILE) | tar -C $(PKG_BUILD_DIR) $(TAR_OPTIONS)
 endef
 
+define Download/nginx-njs
+  VERSION:=376b540b9692c61c1a17b78c65192287b60d8e15
+  SUBDIR:=nginx-njs
+  FILE:=njs-$$(VERSION).tax.xz
+  URL:=https://github.com/nginx/njs.git
+  MIRROR_HASH:=96984ed32ee1cb617495b177dc646d3ae365f5be93263cd300212aa3a22fad2f
+  PROTO:=git
+endef
+
+define Prepare/nginx-njs
+	$(eval $(Download/nginx-njs))
+	xzcat $(DL_DIR)/$(FILE) | tar -C $(PKG_BUILD_DIR) $(TAR_OPTIONS)
+endef
+
 define Download/nginx-naxsi
   VERSION:=951123ad456bdf5ac94e8d8819342fe3d49bc002
   SUBDIR:=nginx-naxsi
@@ -562,6 +596,11 @@ ifeq ($(CONFIG_NGINX_TS_MODULE),y)
   $(Prepare/nginx-ts)
 endif
 
+ifeq ($(CONFIG_NGINX_NJS_MODULE),y)
+  $(eval $(call Download,nginx-njs))
+  $(Prepare/nginx-njs)
+endif
+
 ifneq "$(or $(CONFIG_NGINX_DAV),$(QUILT))" ""
   $(eval $(call Download,nginx-dav-ext-module))
   $(Prepare/nginx-dav-ext-module)
@@ -578,6 +617,7 @@ endef
 $(eval $(call BuildPackage,nginx-ssl))
 $(eval $(call BuildPackage,nginx-all-module))
 $(eval $(call BuildPackage,nginx-mod-luci))
+$(eval $(call BuildPackage,nginx-mod-njs))
 
 # TODO: remove after a transition period (together with pkg nginx-util):
 # It is for smoothly substituting nginx and nginx-mod-luci-ssl (by nginx-ssl
