diff --git a/Makefile b/Makefile
index fd1a216..eb76291 100644
--- a/Makefile
+++ b/Makefile
@@ -47,7 +47,9 @@ $(call Package/wget/Default)
   TITLE+= (with SSL support)
   VARIANT:=ssl
   PROVIDES+=wget-ssl
-  ALTERNATIVES:=300:/usr/bin/wget:/usr/libexec/wget-ssl
+  ALTERNATIVES:=\
+    300:/usr/bin/wget:/usr/libexec/wget-ssl \
+    300:/usr/bin/wget-ssl:/usr/libexec/wget-ssl
 endef
 
 define Package/wget/description
@@ -60,7 +62,9 @@ $(call Package/wget/Default)
   TITLE+= (without SSL support)
   VARIANT:=nossl
   PROVIDES+=wget
-  ALTERNATIVES:=300:/usr/bin/wget:/usr/libexec/wget-nossl
+  ALTERNATIVES:=\
+    300:/usr/bin/wget:/usr/libexec/wget-nossl \
+    300:/usr/bin/wget-nossl:/usr/libexec/wget-nossl
 endef
 
 define Package/wget-nossl/description
