--- a/htdocs/luci-static/resources/view/ddns/overview.js
+++ b/htdocs/luci-static/resources/view/ddns/overview.js
@@ -179,8 +179,10 @@ return view.extend({
 
 	handleToggleDDns: function(m, ev) {
 		return this.callInitAction('ddns', 'enable')
-			.then(L.bind(function (action) { return this.callInitAction('ddns', action ? 'disable' : 'enable')}, this))
-			.then(L.bind(function (action) { return this.callInitAction('ddns', action ? 'stop' : 'start')}, this))
+			.then(L.bind(function (action) {
+				this.callInitAction('ddns', action ? 'disable' : 'enable')
+				this.callInitAction('ddns', action ? 'stop' : 'start')
+			}, this))
 			.then(L.bind(m.render, m))
 			.catch(function(e) { ui.addNotification(null, E('p', e.message)) });
 	},
