diff --git a/include/image.mk b/include/image.mk
index de08d04..bc64e60 100644
--- a/include/image.mk
+++ b/include/image.mk
@@ -231,7 +231,7 @@ define Image/mkfs/squashfs-common
 	$(STAGING_DIR_HOST)/bin/mksquashfs4 $(call mkfs_target_dir,$(1)) $@ \
 		-nopad -noappend -root-owned \
 		-comp $(SQUASHFSCOMP) $(SQUASHFSOPT) \
-		-processors 1
+		-processors $(shell nproc)
 endef
 
 ifeq ($(CONFIG_TARGET_ROOTFS_SECURITY_LABELS),y)
diff --git a/include/netfilter.mk b/include/netfilter.mk
index 77b4c6c..aa97165 100644
--- a/include/netfilter.mk
+++ b/include/netfilter.mk
@@ -107,6 +107,19 @@ $(eval $(call nf_add,IPT_FILTER,CONFIG_NETFILTER_XT_MATCH_STRING, $(P_XT)xt_stri
 $(eval $(call nf_add,IPT_FILTER,CONFIG_NETFILTER_XT_MATCH_BPF, $(P_XT)xt_bpf))
 
 
+# imq
+
+$(eval $(call nf_add,IPT_IMQ,CONFIG_IP_NF_TARGET_IMQ, $(P_V4)ipt_IMQ))
+$(eval $(call nf_add,IPT_IMQ,CONFIG_NETFILTER_XT_TARGET_IMQ, $(P_XT)xt_IMQ))
+
+# gargoyle-qos
+
+$(eval $(call nf_add,IPT_BANDWIDTH,CONFIG_IP_NF_MATCH_BANDWIDTH, $(P_V4)ipt_bandwidth))
+$(eval $(call nf_add,IPT_TIMERANGE,CONFIG_IP_NF_MATCH_TIMERANGE, $(P_V4)ipt_timerange))
+$(eval $(call nf_add,IPT_WEBMON,CONFIG_IP_NF_MATCH_WEBMON, $(P_V4)ipt_webmon))
+$(eval $(call nf_add,IPT_WEBURL,CONFIG_IP_NF_MATCH_WEBURL, $(P_V4)ipt_weburl))
+
+
 # ipopt
 
 $(eval $(call nf_add,IPT_IPOPT,CONFIG_NETFILTER_XT_MATCH_DSCP, $(P_XT)xt_dscp))
@@ -146,6 +159,10 @@ $(eval $(call nf_add,IPT_IPSEC,CONFIG_NETFILTER_XT_MATCH_POLICY, $(P_XT)xt_polic
 # flow offload support
 $(eval $(call nf_add,IPT_FLOW,CONFIG_NETFILTER_XT_TARGET_FLOWOFFLOAD, $(P_XT)xt_FLOWOFFLOAD))
 
+# cgroup
+
+$(eval $(call nf_add,IPT_CGROUP,CONFIG_NETFILTER_XT_MATCH_CGROUP, $(P_XT)xt_cgroup))
+
 # IPv6
 
 # kernel only
diff --git a/include/target.mk b/include/target.mk
index a2e76f8..8451582 100644
--- a/include/target.mk
+++ b/include/target.mk
@@ -28,7 +28,39 @@ DEFAULT_PACKAGES:=\
 	uci \
 	uclient-fetch \
 	urandom-seed \
-	urngd
+	urngd \
+	netifd \
+	coremark \
+	coreutils \
+	kmod-nf-nathelper \
+	kmod-nf-nathelper-extra \
+	kmod-ipt-raw \
+	wget \
+	ca-certificates \
+	default-settings \
+	luci \
+	luci-lib-base \
+	luci-compat \
+	luci-lib-fs \
+	luci-lib-ipkg \
+	luci-proto-relay \
+	luci-app-ddns \
+	luci-app-sqm \
+	luci-app-upnp \
+	luci-app-adbyby-plus \
+	luci-app-autoreboot \
+	luci-app-vsftpd \
+	luci-app-ssr-plus \
+	luci-app-arpbind \
+	luci-app-vlmcsd \
+	luci-app-wol \
+	luci-app-ramfree \
+	luci-app-flowoffload \
+	luci-app-nlbwmon \
+	luci-app-accesscontrol \
+	ddns-scripts_aliyun \
+	ddns-scripts_dnspod \
+	bind-host
 
 ifneq ($(CONFIG_SELINUX),)
 DEFAULT_PACKAGES+=busybox-selinux procd-selinux
@@ -43,14 +75,15 @@ DEFAULT_PACKAGES.nas:=\
 	block-mount \
 	fdisk \
 	lsblk \
-	mdadm
+	mdadm \
+	automount \
+	autosamba
 # For router targets
 DEFAULT_PACKAGES.router:=\
-	dnsmasq \
+	dnsmasq-full \
 	firewall \
 	ip6tables \
 	iptables \
-	kmod-ipt-offload \
 	odhcp6c \
 	odhcpd-ipv6only \
 	ppp \
