diff --git a/netfilter.mk b/netfilter.mk
index 803749d..6b58776 100644
--- a/netfilter.mk
+++ b/netfilter.mk
@@ -106,6 +106,18 @@ $(eval $(call nf_add,IPT_FILTER,CONFIG_NETFILTER_XT_MATCH_STRING, $(P_XT)xt_stri
 $(eval $(call nf_add,IPT_FILTER,CONFIG_NETFILTER_XT_MATCH_BPF, $(P_XT)xt_bpf))
 
 
+# imq
+
+$(eval $(call nf_add,IPT_IMQ,CONFIG_IP_NF_TARGET_IMQ, $(P_V4)ipt_IMQ))
+$(eval $(call nf_add,IPT_IMQ,CONFIG_NETFILTER_XT_TARGET_IMQ, $(P_XT)xt_IMQ))
+
+# gargoyle-qos
+
+$(eval $(call nf_add,IPT_BANDWIDTH,CONFIG_IP_NF_MATCH_BANDWIDTH, $(P_V4)ipt_bandwidth))
+$(eval $(call nf_add,IPT_TIMERANGE,CONFIG_IP_NF_MATCH_TIMERANGE, $(P_V4)ipt_timerange))
+$(eval $(call nf_add,IPT_WEBMON,CONFIG_IP_NF_MATCH_WEBMON, $(P_V4)ipt_webmon))
+$(eval $(call nf_add,IPT_WEBURL,CONFIG_IP_NF_MATCH_WEBURL, $(P_V4)ipt_weburl))
+
 # ipopt
 
 $(eval $(call nf_add,IPT_IPOPT,CONFIG_NETFILTER_XT_MATCH_DSCP, $(P_XT)xt_dscp))
diff --git a/target.mk b/target.mk
index a2e76f8..8451582 100644
--- a/target.mk
+++ b/target.mk
@@ -17,7 +17,7 @@ DEFAULT_PACKAGES:=\
 	fstools \
 	libc \
 	libgcc \
-	libustream-wolfssl \
+	libustream-openssl \
 	logd \
 	mtd \
 	netifd \
@@ -25,7 +25,37 @@ DEFAULT_PACKAGES:=\
 	uci \
 	uclient-fetch \
 	urandom-seed \
-	urngd
+	urngd \
+	netifd \
+	coremark \
+	coreutils \
+	kmod-nf-nathelper \
+	kmod-nf-nathelper-extra \
+	kmod-ipt-raw \
+	wget-ssl \
+	ca-certificates \
+	default-settings \
+	luci \
+	luci-lib-base \
+	luci-compat \
+	luci-lib-fs \
+	luci-lib-ipkg \
+	luci-proto-relay \
+	luci-app-ddns \
+	luci-app-sqm \
+	luci-app-upnp \
+	luci-app-autorebootschedule \
+	luci-app-vsftpd \
+	luci-app-ssr-plus-mosdns \
+	luci-app-arpbind \
+	luci-app-control-vlmcsd \
+	luci-app-wol \
+	luci-app-ramfree \
+	luci-app-nlbwmon \
+	luci-app-accesscontrol \
+	ddns-scripts_aliyun \
+	ddns-scripts_dnspod \
+	bind-host
 
 ifneq ($(CONFIG_SELINUX),)
 DEFAULT_PACKAGES+=busybox-selinux procd-selinux
@@ -50,13 +80,15 @@ DEFAULT_PACKAGES.nas:=\
 	block-mount \
 	fdisk \
 	lsblk \
-	mdadm
+	mdadm \
+	automount \
+	autosamba
 # For router targets
 DEFAULT_PACKAGES.router:=\
-	dnsmasq \
-	firewall4 \
-	nftables \
-	kmod-nft-offload \
+	dnsmasq-full \
+	firewall \
+	ip6tables \
+	iptables \
 	odhcp6c \
 	odhcpd-ipv6only \
 	ppp \
