--- a/netfilter.mk
+++ b/netfilter.mk
@@ -103,6 +103,19 @@ $(eval $(call nf_add,IPT_FILTER,CONFIG_NETFILTER_XT_MATCH_STRING, $(P_XT)xt_stri
 $(eval $(call nf_add,IPT_FILTER,CONFIG_NETFILTER_XT_MATCH_BPF, $(P_XT)xt_bpf))
 
 
+# imq
+
+$(eval $(call nf_add,IPT_IMQ,CONFIG_IP_NF_TARGET_IMQ, $(P_V4)ipt_IMQ))
+$(eval $(call nf_add,IPT_IMQ,CONFIG_NETFILTER_XT_TARGET_IMQ, $(P_XT)xt_IMQ))
+
+# gargoyle-qos
+
+$(eval $(call nf_add,IPT_BANDWIDTH,CONFIG_IP_NF_MATCH_BANDWIDTH, $(P_V4)ipt_bandwidth))
+$(eval $(call nf_add,IPT_TIMERANGE,CONFIG_IP_NF_MATCH_TIMERANGE, $(P_V4)ipt_timerange))
+$(eval $(call nf_add,IPT_WEBMON,CONFIG_IP_NF_MATCH_WEBMON, $(P_V4)ipt_webmon))
+$(eval $(call nf_add,IPT_WEBURL,CONFIG_IP_NF_MATCH_WEBURL, $(P_V4)ipt_weburl))
+
+
 # ipopt
 
 $(eval $(call nf_add,IPT_IPOPT,CONFIG_NETFILTER_XT_MATCH_DSCP, $(P_XT)xt_dscp))
@@ -139,6 +152,10 @@ $(eval $(call nf_add,IPT_IPSEC,CONFIG_NETFILTER_XT_MATCH_POLICY, $(P_XT)xt_polic
 # flow offload support
 $(eval $(call nf_add,IPT_FLOW,CONFIG_NETFILTER_XT_TARGET_FLOWOFFLOAD, $(P_XT)xt_FLOWOFFLOAD))
 
+# cgroup
+
+$(eval $(call nf_add,IPT_CGROUP,CONFIG_NETFILTER_XT_MATCH_CGROUP, $(P_XT)xt_cgroup))
+
 # IPv6
 
 # kernel only
--- a/target.mk
+++ b/target.mk
@@ -20,19 +20,49 @@ DEVICE_TYPE?=router
 ##
 DEFAULT_PACKAGES:=\
 	base-files \
+	bind-host \
 	ca-bundle \
+	ca-certificates \
+	coremark \
+	coreutils \
+	default-settings \
+	ddns-scripts_aliyun \
+	ddns-scripts_dnspod \
 	dropbear \
 	fstools \
+	kmod-nf-nathelper \
+	kmod-nf-nathelper-extra \
+	kmod-ipt-raw \
 	libc \
 	libgcc \
-	libustream-mbedtls \
+	libustream-openssl \
 	logd \
+	luci \
+	luci-app-accesscontrol \
+	luci-app-arpbind \
+	luci-app-autorebootschedule \
+	luci-app-control-vlmcsd \
+	luci-app-ddns \
+	luci-app-nlbwmon \
+	luci-app-ramfree \
+	luci-app-sqm \
+	luci-app-ssr-plus \
+	luci-app-upnp \
+	luci-app-vsftpd \
+	luci-app-wol \
+	luci-compat \
+	luci-lib-base \
+	luci-lib-fs \
+	luci-lib-ipkg \
+	luci-proto-relay \
 	mtd \
 	netifd \
+	opkg \
 	uci \
 	uclient-fetch \
 	urandom-seed \
-	urngd
+	urngd \
+	wget-ssl
 
 ##@
 # @brief Default packages for @DEVICE_TYPE basic.
@@ -42,6 +72,8 @@ DEFAULT_PACKAGES.basic:=
 # @brief Default packages for @DEVICE_TYPE nas.
 ##
 DEFAULT_PACKAGES.nas:=\
+	automount \
+	autosamba \
 	block-mount \
 	fdisk \
 	lsblk \
@@ -50,10 +82,11 @@ DEFAULT_PACKAGES.nas:=\
 # @brief Default packages for @DEVICE_TYPE router.
 ##
 DEFAULT_PACKAGES.router:=\
-	dnsmasq \
-	firewall4 \
-	nftables \
-	kmod-nft-offload \
+	dnsmasq-full \
+	firewall \
+	iptables \
+	ip6tables \
+	kmod-ipt-offload \
 	odhcp6c \
 	odhcpd-ipv6only \
 	ppp \
