diff --git a/Config.in b/Config.in
index 357589e..9c203b5 100644
--- a/Config.in
+++ b/Config.in
@@ -6,9 +6,16 @@ choice
 	help
 	  Select the version of gcc you wish to use.
 
+	config GCC_USE_VERSION_7
+		bool "gcc 7.x"
+		depends on !arc
+
 	config GCC_USE_VERSION_8
 		bool "gcc 8.x"
 
+	config GCC_USE_VERSION_9
+		bool "gcc 9.x"
+
 	config GCC_USE_VERSION_10
 		bool "gcc 10.x"
 
diff --git a/Config.version b/Config.version
index 61506b6..610f8e0 100644
--- a/Config.version
+++ b/Config.version
@@ -1,13 +1,28 @@
+config GCC_VERSION_7
+	default y if GCC_USE_VERSION_7
+	bool
+
 config GCC_VERSION_8
 	default y if GCC_USE_VERSION_8
 	bool
 
+config GCC_VERSION_9
+	default y if GCC_USE_VERSION_9
+	bool
+
 config GCC_VERSION_10
 	default y if GCC_USE_VERSION_10
 	bool
 
 config GCC_VERSION
 	string
+	default "7.5.0"		if GCC_VERSION_7
 	default "8.4.0"		if GCC_VERSION_8
-	default "10.3.0"	if GCC_VERSION_10
+	default "9.3.0"		if GCC_VERSION_9
+	default "10.3.0"		if GCC_VERSION_10
 	default "11.2.0"
+
+config GCC_USE_IREMAP
+	bool
+	default y if GCC_USE_VERSION_7
+	default n
