--- a/htdocs/luci-static/resources/view/network/dhcp.js
+++ b/htdocs/luci-static/resources/view/network/dhcp.js
@@ -389,6 +389,7 @@
 		s.tab('logging', _('Log'));
 		s.tab('files', _('Files'));
 		s.tab('relay', _('Relay'));
+		//s.tab('custom_domain', _('Custom Redirect Domain'));
 
 		// Begin general
 		s.taboption('general', form.Flag, 'authoritative',
@@ -625,6 +626,26 @@
 		});
 		// End pxe_tftp
 
+		// Begin custom_domain
+		//o = s.taboption('custom_domain', form.SectionValue, 'domain', form.GridSection, 'domain', null,
+			//_('Define a custom domain name and the corresponding PTR record'));
+
+		//ss = o.subsection;
+
+		//ss.addremove = true;
+		//ss.anonymous = true;
+
+		//so = ss.option(form.Value, 'name', _('Domain name'));
+		//so.datatype = 'hostname';
+		//so.rmempty  = true;
+
+		//so = ss.option(form.Value, 'ip', _('<abbr title=\"Internet Protocol Version 4\">IPv4</abbr>-Address'));
+		//so.datatype = 'or(ip4addr,"ignore")';
+
+		//so = ss.option(form.Value, 'comments', _('Comments'));
+		//so.rmempty  = true;
+		// End custom_domain
+
 		return s;
 	},
 
--- a/htdocs/luci-static/resources/view/network/dns.js
+++ b/htdocs/luci-static/resources/view/network/dns.js
@@ -209,6 +209,10 @@
 		s.tab('ipsets', _('IP Sets'));
 
 		// Begin general
+		s.taboption('general', form.Flag, 'dns_redirect',
+			_('DNS Redirect'),
+			_('Force redirect all local DNS queries to DNSMasq, a.k.a. DNS Hijacking.'));
+
 		o = s.taboption('general', form.Value, 'local',
 			_('Resolve these locally'),
 			_('Never forward these matching domains or subdomains; resolve from DHCP or hosts files only.'));
@@ -624,6 +628,16 @@
 		o.optional = false;
 		o.rmempty = false;
 
+		//o = s.taboption('filteropts', form.Flag, 'filter_https',
+			//_('Disable HTTPS DNS Type forwards'),
+			//_('Filter HTTPS DNS Query Type Name Resolve'));
+		//o.optional = true;
+
+		//o = s.taboption('filteropts', form.Flag, 'filter_unknown',
+			//_('Disable Unknown DNS Type forwards'),
+			//_('Filter Unknown DNS Query Type Name Resolve'));
+		//o.optional = true;
+
 		o = s.taboption('filteropts', form.Flag, 'boguspriv',
 			_('Filter private'),
 			customi18n(
@@ -646,6 +660,7 @@
 			_('Filter IPv4 A records'),
 			_('Remove IPv4 addresses from the results and only return IPv6 addresses.'));
 		o.optional = true;
+		o.depends('filter_aaaa', '0');
 
 		o = s.taboption('filteropts', form.MultiValue, 'filter_rr',
 			_('Filter arbitrary RR'), _('Removes records of the specified type(s) from answers.'));
@@ -757,6 +772,13 @@
 		o.placeholder = 3600;
 		// End limits
 
+		//o = s.taboption('limits', form.Value, 'mini_ttl',
+			//_('Minimum TTL to send to clients'),
+			//_('Modify DNS entries minimum TTL (max is 86400, 0 is no modify)'));
+		//o.optional = true;
+		//o.datatype = 'range(0,86400)';
+		//o.placeholder = 0;
+
 		// Being logging
 		o = s.taboption('logging', form.Flag, 'logqueries',
 			_('Log queries'),
