diff --git a/htdocs/luci-static/resources/view/status/include/30_network.js b/htdocs/luci-static/resources/view/status/include/30_network.js
index 1a0f703..4a39e22 100644
--- a/htdocs/luci-static/resources/view/status/include/30_network.js
+++ b/htdocs/luci-static/resources/view/status/include/30_network.js
@@ -1,8 +1,14 @@
 'use strict';
 'require baseclass';
 'require fs';
+'require rpc';
 'require network';
 
+var callOnlineUsers = rpc.declare({
+        object: 'luci',
+        method: 'getOnlineUsers'
+});
+
 function progressbar(value, max, byte) {
 	var vn = parseInt(value) || 0,
 	    mn = parseInt(max) || 100,
@@ -67,7 +73,8 @@ return baseclass.extend({
 			fs.trimmed('/proc/sys/net/netfilter/nf_conntrack_count'),
 			fs.trimmed('/proc/sys/net/netfilter/nf_conntrack_max'),
 			network.getWANNetworks(),
-			network.getWAN6Networks()
+			network.getWAN6Networks(),
+			L.resolveDefault(callOnlineUsers(), {})
 		]);
 	},
 
@@ -75,21 +82,32 @@ return baseclass.extend({
 		var ct_count  = +data[0],
 		    ct_max    = +data[1],
 		    wan_nets  = data[2],
-		    wan6_nets = data[3];
+		    wan6_nets = data[3],
+		    onlineusers = data[4];
 
 		var fields = [
-			_('Active Connections'), ct_max ? ct_count : null
+			_('Active Connections'), ct_max ? ct_count : null,
+			_('Online Users'), onlineusers ? onlineusers.onlineusers : null
 		];
 
 		var ctstatus = E('table', { 'class': 'table' });
 
 		for (var i = 0; i < fields.length; i += 2) {
-			ctstatus.appendChild(E('tr', { 'class': 'tr' }, [
-				E('td', { 'class': 'td left', 'width': '33%' }, [ fields[i] ]),
-				E('td', { 'class': 'td left' }, [
-					(fields[i + 1] != null) ? progressbar(fields[i + 1], ct_max) : '?'
-				])
-			]));
+			if (fields[i] == _('Online Users')) {
+				ctstatus.appendChild(E('tr', { 'class': 'tr' }, [
+					E('td', { 'class': 'td left', 'width': '33%' }, [ fields[i] ]),
+					E('td', { 'class': 'td left' }, [
+						(fields[i + 1] != null) ? fields[i + 1] : '?'
+					])
+				]));
+			} else {
+				ctstatus.appendChild(E('tr', { 'class': 'tr' }, [
+					E('td', { 'class': 'td left', 'width': '33%' }, [ fields[i] ]),
+					E('td', { 'class': 'td left' }, [
+						(fields[i + 1] != null) ? progressbar(fields[i + 1], ct_max) : '?'
+					])
+				]));
+			}
 		}
 
 		var netstatus = E('div', { 'class': 'network-status-table' });
