diff --git a/htdocs/luci-static/resources/view/firewall/zones.js b/htdocs/luci-static/resources/view/firewall/zones.js
index 58cd2a7..aa85bd4 100644
--- a/htdocs/luci-static/resources/view/firewall/zones.js
+++ b/htdocs/luci-static/resources/view/firewall/zones.js
@@ -56,6 +56,7 @@ return view.extend({
 		};
 
 		o = s.option(form.Flag, 'drop_invalid', _('Drop invalid packets'));
+		o = s.option(form.Flag, 'fullcone', _('Enable FullCone NAT'));
 
 		var p = [
 			s.option(form.ListValue, 'input', _('Input')),
diff --git a/po/zh_Hans/firewall.po b/po/zh_Hans/firewall.po
index 6ab67e1..f320484 100644
--- a/po/zh_Hans/firewall.po
+++ b/po/zh_Hans/firewall.po
@@ -303,6 +303,9 @@ msgstr "启用 NAT 环回"
 msgid "Enable SYN-flood protection"
 msgstr "启用 SYN-flood 防御"
 
+msgid "Enable FullCone NAT"
+msgstr "启用FullCone-NAT"
+
 #: applications/luci-app-firewall/htdocs/luci-static/resources/view/firewall/zones.js:264
 msgid "Enable logging on this zone"
 msgstr "启用此区域的日志记录"
