diff --git a/Makefile b/Makefile
index d936152..6272b0d 100644
--- a/Makefile
+++ b/Makefile
@@ -31,7 +31,7 @@ include $(INCLUDE_DIR)/package.mk
 define Package/netdata
   SECTION:=admin
   CATEGORY:=Administration
-  DEPENDS:=+zlib +libuuid +libuv +libmnl +libjson-c
+  DEPENDS:=+zlib +libuuid +libuv +libmnl +libjson-c +libopenssl
   TITLE:=Real-time performance monitoring tool
   URL:=https://www.netdata.cloud/
 endef
@@ -55,7 +55,6 @@ CONFIGURE_ARGS += \
 	--enable-lto \
 	--disable-ebpf \
 	--without-libcap \
-	--disable-https \
 	--disable-dbengine \
 	--disable-compression \
 	--disable-plugin-nfacct \
diff --git a/files/netdata.conf b/files/netdata.conf
index 4a247da..246d0f0 100644
--- a/files/netdata.conf
+++ b/files/netdata.conf
@@ -12,6 +12,7 @@
 	error log = syslog
 	access log = none
 	run as user = root
+	plugins directory = "/usr/lib/netdata/plugins.d" "/etc/netdata/custom-plugins.d"
 
 [web]
 	allow connections from = localhost 10.* 192.168.* 172.16.* 172.17.* 172.18.* 172.19.* 172.20.* 172.21.* 172.22.* 172.23.* 172.24.* 172.25.* 172.26.* 172.27.* 172.28.* 172.29.* 172.30.* 172.31.*
@@ -25,6 +26,27 @@
 	node.d = no
 	python.d = no
 
+
+[plugin:proc]
+        ipc =no
+        /proc/sysvipc/shm = no
+        /sys/devices/system/edac/mc = no
+        /sys/devices/system/node = no
+        /proc/net/sockstat = no
+        /proc/net/netstat = no
+        /proc/net/snmp = no
+        /proc/net/softnet_stat = no
+        /proc/net/sctp/snmp = no
+        /proc/net/ip_vs/stats = no
+        /proc/net/stat/synproxy = no
+        /sys/kernel/mm/ksm = no
+        /dev/mapper = no
+	semaphore totals = no
+
+[plugin:proc:/proc/diskstats]
+        path to /dev/vx/dsk  =
+        path to /dev/disk/by-label =</font>
+
 [health]
 	enabled = no
 
