diff --git a/image/mt7621.mk b/image/mt7621.mk
index 0315822..30b944f 100644
--- a/image/mt7621.mk
+++ b/image/mt7621.mk
@@ -275,8 +275,8 @@ define Device/d-team_newifi-d2
   IMAGE_SIZE := 32448k
   DEVICE_VENDOR := Newifi
   DEVICE_MODEL := D2
-  DEVICE_PACKAGES := kmod-mt7603 kmod-mt76x2 kmod-usb3 \
-	kmod-usb-ledtrig-usbport wpad-basic
+  DEVICE_PACKAGES := kmod-mt7603e kmod-mt76x2e kmod-usb3 \
+	kmod-usb-ledtrig-usbport luci-app-mtwifi
 endef
 TARGET_DEVICES += d-team_newifi-d2
 
@@ -849,10 +849,8 @@ define Device/phicomm_k2p
   IMAGE_SIZE := 15744k
   DEVICE_VENDOR := Phicomm
   DEVICE_MODEL := K2P
-  DEVICE_ALT0_VENDOR := Phicomm
-  DEVICE_ALT0_MODEL := KE 2P
   SUPPORTED_DEVICES += k2p
-  DEVICE_PACKAGES := kmod-mt7615e kmod-mt7615-firmware wpad-basic
+  DEVICE_PACKAGES := kmod-mt7615d luci-app-mtwifi
 endef
 TARGET_DEVICES += phicomm_k2p
 
@@ -1032,7 +1030,7 @@ define Device/xiaomi-ac2100
   IMAGE/rootfs0.bin := append-ubi | check-size
   IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
   DEVICE_VENDOR := Xiaomi
-  DEVICE_PACKAGES := kmod-mt7603 kmod-mt7615e kmod-mt7615-firmware wpad-basic uboot-envtools
+  DEVICE_PACKAGES := kmod-mt7603e kmod-mt7615d luci-app-mtwifi uboot-envtools
 endef
 
 define Device/xiaomi_mi-router-ac2100
diff --git a/mt7621/base-files/etc/board.d/01_leds b/mt7621/base-files/etc/board.d/01_leds
index 43cd77c..6ce8d1f 100755
--- a/mt7621/base-files/etc/board.d/01_leds
+++ b/mt7621/base-files/etc/board.d/01_leds
@@ -19,8 +19,8 @@ led_wlan="$(get_dt_led wlan)"
 case $board in
 d-team,newifi-d2)
 	ucidef_set_led_netdev "internet" "internet" "$boardname:amber:internet" "wan"
-	ucidef_set_led_netdev "wlan2g" "WiFi 2.4GHz" "$boardname:blue:wlan2g" "wlan0"
-	ucidef_set_led_netdev "wlan5g" "WiFi 5GHz" "$boardname:blue:wlan5g" "wlan1"
+	ucidef_set_led_netdev "wlan2g" "WiFi 2.4GHz" "$boardname:blue:wlan2g" "ra0"
+	ucidef_set_led_netdev "wlan5g" "WiFi 5GHz" "$boardname:blue:wlan5g" "rai0"
 	;;
 d-team,pbr-m1|\
 gehua,ghl-r-001)
diff --git a/mt7621/base-files/etc/board.d/02_network b/mt7621/base-files/etc/board.d/02_network
index 9c32f59..8d136a5 100755
--- a/mt7621/base-files/etc/board.d/02_network
+++ b/mt7621/base-files/etc/board.d/02_network
@@ -19,6 +19,9 @@ ramips_setup_interfaces()
 	xiaomi,mir3g-v2)
 		ucidef_set_interfaces_lan_wan "lan1 lan2" "wan"
 		;;
+	d-team,newifi-d2)
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4 ra0 rai0" "wan"
+		;;
 	edimax,re23s|\
 	mikrotik,routerboard-m11g|\
 	netgear,ex6150|\
@@ -34,7 +37,7 @@ ramips_setup_interfaces()
 	xiaomi,mi-router-ac2100|\
 	xiaomi,mir3p|\
 	xiaomi,redmi-router-ac2100)
-		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3" "wan"
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 ra0 rai0" "wan"
 		;;
 	gnubee,gb-pc1|\
 	gnubee,gb-pc2)
@@ -47,6 +50,9 @@ ramips_setup_interfaces()
 	mikrotik,routerboard-750gr3)
 		ucidef_set_interfaces_lan_wan "lan2 lan3 lan4 lan5" "wan"
 		;;
+	phicomm,k2p)
+		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4 ra0 rax0" "wan"
+		;;
 	ubnt,edgerouter-x)
 		ucidef_set_interface_lan_wan "eth1 eth2 eth3 eth4" "eth0"
 		;;
