--- a/Makefile
+++ b/Makefile
@@ -333,6 +333,8 @@ define Package/ddns-scripts/postinst
 if [ -z "$${IPKG_INSTROOT}" ]; then
 	/etc/init.d/ddns enabled
 	/etc/init.d/ddns start
+
+	grep -q "/etc/init.d/ddns" "/etc/rc.local" || sed -i "2 a /etc/init.d/ddns restart" "/etc/rc.local"
 fi
 exit 0
 endef
@@ -342,6 +344,8 @@ define Package/ddns-scripts/prerm
 if [ -n "$${IPKG_INSTROOT}" ]; then
 	/etc/init.d/ddns stop
 	/etc/init.d/ddns disable
+
+	sed -i "/ddns restart/d" "/etc/rc.local"
 fi
 exit 0
 endef
